### PR TITLE
Added asserts for things

### PR DESCRIPTION
**Pull request by StephenHodgson**
- Added null check assert for main camera tag in `CameraCache.Main`.
    - (This should probably be backported into master as well.)
- Added asserts in Gaze and Focus Providers for MR Manager and Input System.

### CODE REVIEW

File: Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs [hunk 1]
```diff
@@ -228,6 +228,9 @@ private void Awake()
 
         private void Start()
         {
+            Debug.Assert(MixedRealityManager.IsInitialized, "No Mixed Reality Manager found in the scene.  Be sure to run the Mixed Reality Configuration.");
+            Debug.Assert(InputSystem != null, "No Input System found, Did you set it up in your configuration profile?");
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs]: Is this really an assertable condition? Arent we building in the flexibility for NOT using the input system?

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs]: > Arent we building in the flexibility for NOT using the input system?

We did, but in this case this is a component in _our_ implementation of the input system.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs]: > Is this really an assertable condition?

And yes.

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs]: Then is it even possible to get into this state? Sorry if I am being dense today :)

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs]: Never. unless someone puts this component on by hand (Which shouldn't happen as we do it for them)

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/InputSystem/Focus/FocusProvider.cs]: Thanks for clarifying... looks good to me.

**Pull request merged**